s3/replication/common/experiments/kafka_consumer.py

```diff
@@ -0,0 +1,158 @@
+       }
+    '''
+    job_id_to_offset = {}
+    last_commited_offset = 0
+
+    def __init__(self):
+        """Initialize."""
+        # List store topic, partition and offset of each job
+        self._consumer_details = {}
+        self._job_id = 0
+
+    # Initialize the kafka consumer
+    # This method should be called from app.py class on startup.
+    def initialize(self, app):
+        self._app = app
+        KafkaMain.consumer = AIOKafkaConsumer(
+            'FDMI',
+            bootstrap_servers='127.0.0.1:9092',
+            group_id="test",
+            auto_offset_reset='latest',
+            enable_auto_commit=False)
+
+    # Getter methods
+    def get_topic(self, job_id):
+        """Returns the topic."""
+        return KafkaMain.job_id_to_offset[job_id]["Topic"]
+
+    def get_partition(self, job_id):
+        return KafkaMain.job_id_to_offset[job_id]["Partition"]
+
+    def get_offset(self, job_id):
+        return KafkaMain.job_id_to_offset[job_id]["Offset"]
+
+    # Remove job from job-id to offset mapping after commit
+    def remove_job_from_mapping(self, job_id):
+        removed_key = KafkaMain.job_id_to_offset.pop(job_id, None)
+        if removed_key is not None:
+            print("Removed job id from mapping : {}".format(removed_key))
+        else:
+            print("Job id [{}] does not exist".format(job_id))
+
+    # Send job for replication after consuming from kafka
+    async def sendJob(self, data):
+        print("Sending Jobs...")
+        fdmi_dict = literal_eval(data)
+        job_record = PrepareReplicationJob.from_fdmi(fdmi_dict)
+
+        if job_record is None:
+            print("BadRequest")
+            return
+
+        jobs_list = self._app['all_jobs']
+        if jobs_list.is_job_present(job_record["replication-id"]):
+            # Duplicate job.
+            print('Duplicate Job!')
+            return
+
+        job = jobs_list.add_job_using_json(job_record)
+        self._job_id = job.get_job_id()
+        print("Added Job : {}".format(self._job_id))
+
+        # Add job id and consumer details list to dictionary
+        KafkaMain.job_id_to_offset[self._job_id] = self._consumer_details
+        print("Job-id to offset mapping : {}".format(KafkaMain.job_id_to_offset))
+        self._consumer_details = {}
+
+    # Start the consumer and add the consumer details into list
+    async def consume(self):
+        await KafkaMain.consumer.start()
+        topic_partition = TopicPartition('FDMI', 0)
+        position = await KafkaMain.consumer.position(topic_partition)
+
+        try:
+            print("In consumer loop")
+            # Consume messages
+            async for msg in KafkaMain.consumer:
+                if msg.offset == position:
+                    pass
+                else:
+                    obj = json.loads(msg.value.decode())
+                    print("Event consumed : {}", obj['cr_val'])
+
+                    self._consumer_details["Topic"] = msg.topic
```

Comment:
I don't think we need such a variable `_consumer_details`.

This dict variable is only used here, and then in sendJob() function.
We can pass the data to sendJob().
So, the `_consumer_details` can be removed from the class member.

Reply:
done